src/world.rs

```diff
@@ -210,8 +225,33 @@ impl World {
 
     pub fn apply_action(&mut self, action: &Action) {
         match action {
-            Action::Destroy { at } => self.destroy_cube(at.clone()),
-            Action::Add { at, block } => self.add_cube(at.clone(), block.clone())
+            Action::Destroy { at } => {
+                let cubes_to_add = self.destroy_cube(at.clone());
+                self.cubes_to_draw.destroy_cube(&at);
+                for cube in cubes_to_add {
+                    self.cubes_to_draw.add_cube(&cube)
+                }
+
+            },
+            Action::Add { at, block } => {
+                let (cubes_to_destroy, cube) = self.add_cube(at.clone(), block.clone());
+                self.cubes_to_draw.add_cube(&cube);
+                for position in cubes_to_destroy {
+                    self.cubes_to_draw.destroy_cube(&position)
+                }
+            }
+        }
+    }
```

Comment:
After changing `cubes_to_draw` to an Option, you will have to change this function so that the extra logic is only done when the option is set to something.

Also, please add some comments that explains a little bit why we have to add cubes when deleting, as it could be confusing. The something like `// Destroying a cube can reveal other cubes to draw`. 

Actually `cube_to_add` is not a good name.
- First there is a typo, it is `cubes_to_add` 
- But I think that `revealed_cubes` is even better, because it allows to not put a comment.